modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EVariable.java

```diff
@@ -43,27 +45,52 @@ public EVariable(Location location, String name) {
 
     @Override
     Output analyze(ClassNode classNode, ScriptRoot scriptRoot, Scope scope, Input input) {
-        if (input.read == false && input.write == false) {
-            throw createError(new IllegalArgumentException("not a statement: variable [" + name + "] not used"));
-        }
-
         Output output = new Output();
+        Class<?> type = scriptRoot.getPainlessLookup().canonicalTypeNameToType(name);
```

Comment:
Is there any possibility of a partial name triggering `type != null`? eg. `java.util.HashMap.abc123`.

Reply:
Tests should cover all use cases of this particular corner case. This should always result in a compiler error - "not a symbol"